src/control/lib/hardware/hwloc/provider.go

```diff
@@ -130,58 +132,120 @@ func (p *Provider) getNUMANodes(topo *topology) (map[uint]*hardware.NUMANode, er
 		}
 		prevNode = numaObj
 
-		nodeStr := numaObj.nodeSet().String()
-
 		id := numaObj.osIndex()
 		newNode := &hardware.NUMANode{
-			ID:       id,
-			NumCores: coresByNode[nodeStr],
-		}
-
-		if devs, exists := pciDevsByNode[id]; exists {
-			newNode.Devices = devs
-		} else {
-			newNode.Devices = make(hardware.PCIDevices)
+			ID:  id,
+			Set: numaObj.nodeSet().String(),
```

Comment:
I guess I wasn't really thinking about this before, but I'm not sure about exposing this string in the NUMANode structure. I'm not convinced this has any meaning outside of hwloc -- it's essentially a secondary identifier for the NUMA node.

If the intention is to create a flag that indicates whether the NUMA node is real or not, I'd be more in favor of either creating a formal synthetic NUMA ID (max_int or something), or having a flag on the NUMA node object itself indicating that it was not actually detected.

Reply:
Hmm. In hindsight, I am not sure why I thought I needed the Set field. I seem to recall that I added when I add support for gathering the cores, and maybe I didn't quite understand how the other stuff worked. I just removed Set and all of the tests still pass, so I'll push an update for this.